src/libs/actions/Report.js

```diff
@@ -616,6 +616,197 @@ function createOptimisticChatReport(participantList) {
     };
 }
 
+/**
+ * @param {String} policyID
+ * @param {String} policyName
+ * @returns {Object}
+ */
+function createOptimisticWorkspaceChats(policyID, policyName) {
+    const announceChatReportID = ReportUtils.generateReportID();
+    const announceChatData = {
+        chatType: CONST.REPORT.CHAT_TYPE.POLICY_ANNOUNCE,
+        policyID,
+        reportID: announceChatReportID,
+        reportName: CONST.REPORT.WORKSPACE_CHAT_ROOMS.ANNOUNCE,
+        pendingAction: CONST.RED_BRICK_ROAD_PENDING_ACTION.ADD,
+        hasOutstandingIOU: false,
+        isOwnPolicyExpenseChat: false,
+        isLoadingReportActions: false,
+        isPinned: false,
+        lastActorEmail: '',
+        lastMessageHtml: '',
+        lastMessageText: null,
+        lastReadSequenceNumber: 0,
+        lastMessageTimestamp: 0,
+        lastVisitedTimestamp: 0,
+        maxSequenceNumber: 0,
+        notificationPreference: '',
+        oldPolicyName: policyName,
+        ownerEmail: '__FAKE__',
+        participants: [currentUserEmail],
+        stateNum: 0,
+        statusNum: 0,
+        visibility: undefined,
+    };
+    const announceReportActionData = {
+        0: {
```

Comment:
Is there a reason we are not using an array? Don't think it's a problem, just curious : )

Reply:
Onyx is built to work with objects not arrays - we've been trying to update any current array usage to objects